pages/api/retrieval.ts

```diff
@@ -0,0 +1,112 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import isRequestAllowed from "@lib/middleware/httpRequestAllowed";
+import { logMessage } from "@lib/logger";
+import { DynamoDBClient, QueryCommand, UpdateItemCommand } from "@aws-sdk/client-dynamodb";
+import { BearerTokenPayload } from "@lib/types";
+import validate from "@lib/middleware/bearerToken";
+
+/**
+ * EntryPoint of retrieval Api. It carries a request to the right method
+ * otherwise returns 500 error status code.
+ * @param req
+ * @param res
+ * @param bearerTokenPayload
+ * @returns
+ */
+const retrieval = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  options?: unknown
+): Promise<void> => {
+  try {
+    return await getFormResponses(req, res, options as BearerTokenPayload);
+  } catch (err) {
+    logMessage.error(err);
+    res.status(500).json({ error: "Error on Server Side" });
+  }
+};
+
+/**
+ * This method fetchs form responses up to the maximum number (maxRecords) that was specified.
+ * It required to have a valid bearer token which must carry a valid formID. When all these conditons are met,
+ * Not only it should return form reponses but also mark those responses as retrieved by updating the attribute Retrieved
+ * from 0 to 1.
+ * @param req
+ * @param res
+ */
+export async function getFormResponses(
+  req: NextApiRequest,
+  res: NextApiResponse,
+  bearerTokenPayload: BearerTokenPayload
+): Promise<void> {
+  const { maxRecords } = req.query;
+  if (!maxRecords || parseInt(maxRecords as string) < 0 || parseInt(maxRecords as string) >= 11) {
+    return res.status(400).json({ error: "Invalid paylaod value found maxRecords" });
+  } else {
```

Comment:
If you'd like the `else {}` block could be removed as the `return` in the if statement above short circuits the function.